packages/edit-post/src/hooks/use-post-history.js

```diff
@@ -34,8 +35,31 @@ export default function usePostHistory( initialPostId, initialPostType ) {
 		[ { postId: initialPostId, postType: initialPostType } ]
 	);
 
-	const onSelectPost = useCallback( ( postId, postType ) => {
-		dispatch( { type: 'push', post: { postId, postType } } );
+	const onSelectPost = useCallback( ( params ) => {
+		const currentArgs = getQueryArgs( window.location.href );
+		const currentUrlWithoutArgs = removeQueryArgs(
+			window.location.href,
+			...Object.keys( currentArgs )
+		);
+
+		const newUrl = addQueryArgs( currentUrlWithoutArgs, {
+			post: params.postId,
+			action: 'edit',
+		} );
+
```

Comment:
The alternative to this would be to create another hook in edit-site, but we would be largely just copying the code from `useLink` to do this which seemed like a duplication of code. I like the addition of the `href` in copying the `useLink` return signature as this allows for right click on edit button to open in new tab, which  matches template editing in the site editor.